[Security Solution] Init rules RBAC configuration

xcrzx opened the pull request:
**Epic: https://github.com/elastic/security-team/issues/8799**

Added a new Rules feature that controls access to:

1. Detection rules
2. Alerts
3. Rule exceptions

In the first iteration, Rules feature doesn't allow more granular customization of a user's role as described in https://github.com/elastic/security-team/issues/8799. Granular controls will be extracted from the Rules feature as sub-features in future iterations.

## How to test this PR

The extraction of the Rules feature from SIEM opens several new possibilities to configure roles.

### Rule: none

<img width="740" height="400" alt="image" src="https://github.com/user-attachments/assets/b2fd9932-3627-426c-87ab-937063edec27" />

This role configuration allows access to the rest of Security Solution without access to Detection Rules and Alerts.

- Rules menu item is visible to the users, no Detection rules (SIEM), no Shared exceptions lists items in the menu. Benchmark rules and Migrations should still be accessible.
- Users should not be able to access any rule page directly
- Users cannot access the Alerts page, Alerts menu is hidden
- No coverage page
- Detection rule monitoring dashboard - depends on access to the .kibana-event-log-* - should not be visible to user
- No security rules shown in Stack management -> Rules
- For the security setup guide (Getting started) - no rule setup will be possible

### Rule: read

This role configuration should allow users to read rules and visit any rule pages but without the ability to edit rules, alerts, or exceptions.

The minimal Kibana feature configuration:

<img width="739" height="399" alt="image" src="https://github.com/user-attachments/assets/2ba64792-e426-418b-bcbf-a0765c3cb538" />

Required index privileges:

<img width="1550" height="255" alt="image" src="https://github.com/user-attachments/assets/393249f5-434a-4243-bad3-1073ce52d2c3" />

- Rule details page:
  - No action snoozing option,
  - No edit rule settings button or actions menu
  - “Enable” control is not editable
  - Export should be possible
- In the rules table (Installed rules and Rules Monitoring tables):
  - Bulk actions: only Export option is present
  - “Enable” control is not editable
  - No action snoozing option
  - No edit rule settings button or actions menu
  - No “Create new rule” button
  - No “Import rule” button
- “Add elastic rules” page:
  - No “Install rule” link
  - No “Install all” button
  - No install rule buttons/or greyed out on the rule flyout.
- Rule Updates tab
  - No Update all, individual rule updates
  - On the update flyout - only option to Close, not Update button
- Stack management Rules
  - Should not be possible to modify the security rules from there

### Rule: all

This role configurration should have access to rules, alerts, and exceptions without limitations.

<img width="733" height="403" alt="image" src="https://github.com/user-attachments/assets/9d4a671a-f168-437d-9017-914d857f01bb" />

<img width="1590" height="254" alt="image" src="https://github.com/user-attachments/assets/48d346dd-7e0b-40b9-b3d2-0ab47eb29f43" />

- Ability to create, import, edit, update and delete rules.
- Rules write access, the user will be able to see rules and details and edit all rule details.

### Additional areas to test

- Old role configuration (`siemV3`) should work correctly. Roles created prior to this PR with `siemV3:all` should map to `siemV4:all` + `rules:all`. Roles with `siemV3:read` to `siemV4:read` and `rules:read`.
- Serverless with predefined roles
- Check the AI4SOC tier for regressions

## PR Handoff TODOs

- [ ] Rebase PR on `main`
  - [x] Fixing merge conflicts
  - [x] Switching `v3` -> `v4` and `v4` -> `v5`
  - [x] Carry over `detection_engine.tsx` changes into newly abstracted files implemented [here](https://github.com/elastic/kibana/pull/222457)
  - [ ] Update test mocks
  - [ ] Fix broken tests
- [ ] Align AI4SOC privileges ([related slack thread](https://elastic.slack.com/archives/C06TGUDNXU2/p1757106894670349))
  - [ ] Switch out all link capabilities
- [ ] SIEM migrations update to RULES WRITE (related [issue](https://github.com/elastic/security-team/issues/13832))
  - [ ] Align with threat hunting
  - [ ] Update warning messages and related behavior
- [ ] Figure out how read or crud alerts in `src/platform/packages/shared/kbn-es/src/serverless_resources/project_roles/security/roles.yml` is used
  - [ ] Delete if not used
- [ ] Integration links should be disabled <img width="450" height="210" alt="image" src="https://github.com/user-attachments/assets/f76038e1-9d03-45ed-aae4-203ac5857acd" />
- [ ] Should we allow users to view the rule monitoring dashboard? Or remove the link? <img width="1641" height="163" alt="image" src="https://github.com/user-attachments/assets/6736c6ca-8032-456d-802a-472b96f74436" />
- [ ] Write new tests
  - [ ] Keeping in mind new version pitfalls ([related comment](https://github.com/elastic/kibana/pull/232113#discussion_r2324304567))
  - [ ] Determine testing coverage approach (cypress? FTR? etc.)
  - [ ] ...
- [ ] Manual testing
  - [ ] Potentially reverting migration changes to smoke test "current" behavior ([comment](https://github.com/elastic/kibana/pull/232113#discussion_r2324304567))
  - [ ] ...
- [ ] Update Elasticsearch controller (predefined serverless roles) code to match this PR
  - [ ] Must be done when this PR gets deployed to serverless
  - [ ] https://github.com/elastic/elasticsearch-controller/blob/main/internal/config/roles/security.yaml
  - [ ] https://github.com/elastic/elasticsearch-controller/blob/main/internal/config/roles/security.ai_soc_engine.yaml

Commit:
Separate Rules RBAC from SIEM

xcrzx commented on x-pack/solutions/security/plugins/security_solution/public/common/components/user_privileges/user_privileges_context.tsx:
```diff
@@ -70,22 +73,32 @@ export const UserPrivilegesProvider = ({
+  const rulesPrivileges = useMemo(() => {
+    const rulesCapabilities = kibanaCapabilities.securitySolutionRulesV1;
+    return {
+      read: rulesCapabilities?.[RULES_UI_EDIT] === true,
```
Yea, it should. Thanks 👍

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/335360)
* Commit: bc73b74021dfebc3d79c9feee7c983defac1c7c1

### Failed CI Steps
* [Quick Checks](https://buildkite.com/elastic/kibana-pull-request/builds/335360#01990fa5-02ab-4100-b9d1-8115f61e4c73)

### History
* :broken_heart: [Build #335338](https://buildkite.com/elastic/kibana-pull-request/builds/335338) failed d7cb5afeadba131f16da7d688c95dc36b5c9ec2b

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/335360","number":335360,"commit":"bc73b74021dfebc3d79c9feee7c983defac1c7c1"},{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/335338","number":335338,"commit":"d7cb5afeadba131f16da7d688c95dc36b5c9ec2b"}],"number":335360}
buildkite-pr-comment-->

elasticmachine commented:
## :broken_heart: Build Failed
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/335438)
* Commit: 8a6718232705c9416a9f3c685c3999589ed289a7

### Failed CI Steps
* [Check Types](https://buildkite.com/elastic/kibana-pull-request/builds/335438#01991035-74f0-4faa-82a0-399589521695)

### History
* :broken_heart: [Build #335383](https://buildkite.com/elastic/kibana-pull-request/builds/335383) failed 5cff3e1ae9e7e9afbfeae310d11080e4cf9f0e07
* :broken_heart: [Build #335360](https://buildkite.com/elastic/kibana-pull-request/builds/335360) failed bc73b74021dfebc3d79c9feee7c983defac1c7c1
* :broken_heart: [Build #335338](https://buildkite.com/elastic/kibana-pull-request/builds/335338) failed d7cb5afeadba131f16da7d688c95dc36b5c9ec2b

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/335438","number":335438,"commit":"8a6718232705c9416a9f3c685c3999589ed289a7"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/335383","number":335383,"commit":"5cff3e1ae9e7e9afbfeae310d11080e4cf9f0e07"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/335360","number":335360,"commit":"bc73b74021dfebc3d79c9feee7c983defac1c7c1"},{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/335338","number":335338,"commit":"d7cb5afeadba131f16da7d688c95dc36b5c9ec2b"}],"number":335438}
buildkite-pr-comment-->

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/335729)
* Commit: 8a18e6c794591697fbaf82f1c7080eab7363d996

### Failed CI Steps
* [Scout: [ security / security_solution ] plugin](https://buildkite.com/elastic/kibana-pull-request/builds/335729#01991451-f747-4339-82c9-b81a4cee8020)

### Test Failures
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/335729#01991451-f747-4339-82c9-b81a4cee8020) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/335729/jobs/01991451-f747-4339-82c9-b81a4cee8020/artifacts/0199145a-d7ef-40d6-8a65-c5c71f27435b) Scout: [ security / security_solution ] plugin / serverless-security - Expandable flyout state sync - should test flyout url sync
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/335729#01991451-f747-4339-82c9-b81a4cee8020) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/335729/jobs/01991451-f747-4339-82c9-b81a4cee8020/artifacts/0199145a-d7ef-41cb-a11c-866d8a694ef9) Scout: [ security / security_solution ] plugin / stateful - Expandable flyout state sync - should test flyout url sync

### History
* :broken_heart: [Build #335486](https://buildkite.com/elastic/kibana-pull-request/builds/335486) failed 5dd5bce1f8848ead016e3d4846423c0fdd2a6c78
* :broken_heart: [Build #335438](https://buildkite.com/elastic/kibana-pull-request/builds/335438) failed 8a6718232705c9416a9f3c685c3999589ed289a7
* :broken_heart: [Build #335383](https://buildkite.com/elastic/kibana-pull-request/builds/335383) failed 5cff3e1ae9e7e9afbfeae310d11080e4cf9f0e07
* :broken_heart: [Build #335360](https://buildkite.com/elastic/kibana-pull-request/builds/335360) failed bc73b74021dfebc3d79c9feee7c983defac1c7c1
* :broken_heart: [Build #335338](https://buildkite.com/elastic/kibana-pull-request/builds/335338) failed d7cb5afeadba131f16da7d688c95dc36b5c9ec2b

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/335729","number":335729,"commit":"8a18e6c794591697fbaf82f1c7080eab7363d996"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/335486","number":335486,"commit":"5dd5bce1f8848ead016e3d4846423c0fdd2a6c78"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/335438","number":335438,"commit":"8a6718232705c9416a9f3c685c3999589ed289a7"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/335383","number":335383,"commit":"5cff3e1ae9e7e9afbfeae310d11080e4cf9f0e07"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/335360","number":335360,"commit":"bc73b74021dfebc3d79c9feee7c983defac1c7c1"},{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/335338","number":335338,"commit":"d7cb5afeadba131f16da7d688c95dc36b5c9ec2b"}],"number":335729}
buildkite-pr-comment-->

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/335952)
* Commit: 252bc9b6caf0541c6c6d81368cb1391437f4261a

### Failed CI Steps
* [Osquery Cypress Tests on Serverless #3](https://buildkite.com/elastic/kibana-pull-request/builds/335952#01991552-ca2c-4c67-8bcd-552c86bcaab5)

### Test Failures
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/335952#01991552-ca2c-4c67-8bcd-552c86bcaab5) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/335952/jobs/01991552-ca2c-4c67-8bcd-552c86bcaab5/artifacts/0199158b-4c73-4d59-8451-fa989cf50b28) Osquery Cypress Tests on Serverless #<span></span>3 / Alert Event Details - Response Actions Form "before all" hook for "adds response actions with osquery with proper validation and form values"

### History
* :broken_heart: [Build #335729](https://buildkite.com/elastic/kibana-pull-request/builds/335729) failed 8a18e6c794591697fbaf82f1c7080eab7363d996
* :broken_heart: [Build #335486](https://buildkite.com/elastic/kibana-pull-request/builds/335486) failed 5dd5bce1f8848ead016e3d4846423c0fdd2a6c78
* :broken_heart: [Build #335438](https://buildkite.com/elastic/kibana-pull-request/builds/335438) failed 8a6718232705c9416a9f3c685c3999589ed289a7
* :broken_heart: [Build #335383](https://buildkite.com/elastic/kibana-pull-request/builds/335383) failed 5cff3e1ae9e7e9afbfeae310d11080e4cf9f0e07
* :broken_heart: [Build #335360](https://buildkite.com/elastic/kibana-pull-request/builds/335360) failed bc73b74021dfebc3d79c9feee7c983defac1c7c1

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/335952","number":335952,"commit":"252bc9b6caf0541c6c6d81368cb1391437f4261a"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/335729","number":335729,"commit":"8a18e6c794591697fbaf82f1c7080eab7363d996"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/335486","number":335486,"commit":"5dd5bce1f8848ead016e3d4846423c0fdd2a6c78"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/335438","number":335438,"commit":"8a6718232705c9416a9f3c685c3999589ed289a7"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/335383","number":335383,"commit":"5cff3e1ae9e7e9afbfeae310d11080e4cf9f0e07"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/335360","number":335360,"commit":"bc73b74021dfebc3d79c9feee7c983defac1c7c1"}],"number":335952}
buildkite-pr-comment-->

gergoabraham commented on src/platform/packages/shared/kbn-es/src/serverless_resources/project_roles/security/roles.yml:
```diff
@@ -43,9 +43,9 @@ viewer:
-        - feature_siemV3.read
-        - feature_siemV3.read_alerts
-        - feature_siemV3.endpoint_list_read
+        - feature_siemV4.read
```
hey there,

i'd advice against updating any role definition in the same PR where we add role migration.

the reason is that having both updated in the same PR would mean that automated tests will use the new `siemV4` version, while users in the wild (both in serverless and ESS) will use older siem version for a while, which means they are depending on the correctness of role migration, and we need to thoroughly test that the migration works correctly.

of course this is possible, it's just seems safer to do it separately, as by not updating the roles in this PR, we have one more safety net on the correctness of the migration: all of the existing automated tests.

story time :) for the previous `siemV3` migration, i also added role migrations and updated the roles in the same PR (https://github.com/elastic/kibana/pull/219566), but turned out i did not think of a use case in the migration, and it was just luck that we had a test case running in MKI env using the `elasticsearch-controller` predefined (and not yet updated) roles to find the bug (https://github.com/elastic/kibana/pull/225331).
but as you see, this was just luck, as we have only a small subset of tests running in MKI, while not updating the roles in the same PR would have found in the PR's CI job the missing use case _before merging the pr_.

let me know if you have a question on the above

(this applies to `x-pack/solutions/security/plugins/security_solution/scripts/endpoint/common/roles_users/serverless/es_serverless_resources/roles.yml` and any other yaml/json/ts file that contains roles used for testing.)

xcrzx replied:
That's a good point. Thanks for sharing your experience, @gergoabraham. I will make sure that the app works with the new privileges, then, probably, revert the config changes to make sure that all tests pass.

denar50 commented:
/ci

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/336574)
* Commit: fb1ebadcdddddc33ebec33de7afb13464849ad49

### Failed CI Steps
* [Explore - Security Solution Cypress Tests #1](https://buildkite.com/elastic/kibana-pull-request/builds/336574#019920c8-d2a7-443d-9f6d-d62b94b2f86a)

### History
* :broken_heart: [Build #336287](https://buildkite.com/elastic/kibana-pull-request/builds/336287) failed f9b355ac54ac7b22d74d5ff7d4d6236860e10845
* :broken_heart: [Build #335952](https://buildkite.com/elastic/kibana-pull-request/builds/335952) failed 252bc9b6caf0541c6c6d81368cb1391437f4261a
* :broken_heart: [Build #335729](https://buildkite.com/elastic/kibana-pull-request/builds/335729) failed 8a18e6c794591697fbaf82f1c7080eab7363d996
* :broken_heart: [Build #335486](https://buildkite.com/elastic/kibana-pull-request/builds/335486) failed 5dd5bce1f8848ead016e3d4846423c0fdd2a6c78
* :broken_heart: [Build #335438](https://buildkite.com/elastic/kibana-pull-request/builds/335438) failed 8a6718232705c9416a9f3c685c3999589ed289a7

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/336574","number":336574,"commit":"fb1ebadcdddddc33ebec33de7afb13464849ad49"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/336287","number":336287,"commit":"f9b355ac54ac7b22d74d5ff7d4d6236860e10845"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/335952","number":335952,"commit":"252bc9b6caf0541c6c6d81368cb1391437f4261a"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/335729","number":335729,"commit":"8a18e6c794591697fbaf82f1c7080eab7363d996"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/335486","number":335486,"commit":"5dd5bce1f8848ead016e3d4846423c0fdd2a6c78"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/335438","number":335438,"commit":"8a6718232705c9416a9f3c685c3999589ed289a7"}],"number":336574}
buildkite-pr-comment-->

denar50 commented:
/ci

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/336727)
* Commit: 20d1fe0c6f6f24583a49cc263ced6cdf98b54dc0

### Failed CI Steps
* [FTR Configs #78](https://buildkite.com/elastic/kibana-pull-request/builds/336727#019928ba-8dd5-4b27-8bdb-edf3fd667a50)
* [Explore - Security Solution Cypress Tests #1](https://buildkite.com/elastic/kibana-pull-request/builds/336727#0199289b-dbcf-4685-9c52-b6f963226e2a)

### Test Failures
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/336727#019928ba-8dd5-4b27-8bdb-edf3fd667a50) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/336727/jobs/019928ba-8dd5-4b27-8bdb-edf3fd667a50/artifacts/019928d0-4ae3-4c3b-b1e8-881e3327b2c7) FTR Configs #<span></span>78 / Endpoint plugin spaces support @<span></span>ess @<span></span>skipInServerless, @<span></span>skipInServerlessMKI Endpoint Artifacts space awareness user role backwards compatibility until siemV4 with siemV3 feature version should allow creating a global artifact on endpoint_list list with original privileges all

### History
* :broken_heart: [Build #336574](https://buildkite.com/elastic/kibana-pull-request/builds/336574) failed fb1ebadcdddddc33ebec33de7afb13464849ad49
* :broken_heart: [Build #336287](https://buildkite.com/elastic/kibana-pull-request/builds/336287) failed f9b355ac54ac7b22d74d5ff7d4d6236860e10845
* :broken_heart: [Build #335952](https://buildkite.com/elastic/kibana-pull-request/builds/335952) failed 252bc9b6caf0541c6c6d81368cb1391437f4261a
* :broken_heart: [Build #335729](https://buildkite.com/elastic/kibana-pull-request/builds/335729) failed 8a18e6c794591697fbaf82f1c7080eab7363d996
* :broken_heart: [Build #335486](https://buildkite.com/elastic/kibana-pull-request/builds/335486) failed 5dd5bce1f8848ead016e3d4846423c0fdd2a6c78

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/336727","number":336727,"commit":"20d1fe0c6f6f24583a49cc263ced6cdf98b54dc0"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/336574","number":336574,"commit":"fb1ebadcdddddc33ebec33de7afb13464849ad49"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/336287","number":336287,"commit":"f9b355ac54ac7b22d74d5ff7d4d6236860e10845"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/335952","number":335952,"commit":"252bc9b6caf0541c6c6d81368cb1391437f4261a"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/335729","number":335729,"commit":"8a18e6c794591697fbaf82f1c7080eab7363d996"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/335486","number":335486,"commit":"5dd5bce1f8848ead016e3d4846423c0fdd2a6c78"}],"number":336727}
buildkite-pr-comment-->

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/337977)
* Commit: 0d7c1245c38ddd0fd8894121412f4255972c37fa

### Failed CI Steps
* [Rule Management - Prebuilt Rules Installation - Security Solution Cypress Tests #1](https://buildkite.com/elastic/kibana-pull-request/builds/337977#0199341d-cdf7-4ea4-80e5-2c1c614619fc)

### History
* :broken_heart: [Build #337165](https://buildkite.com/elastic/kibana-pull-request/builds/337165) failed 2f43bf9437336e99f8837aa37a9074ca9e0a534e
* :broken_heart: [Build #336727](https://buildkite.com/elastic/kibana-pull-request/builds/336727) failed 20d1fe0c6f6f24583a49cc263ced6cdf98b54dc0
* :broken_heart: [Build #336574](https://buildkite.com/elastic/kibana-pull-request/builds/336574) failed fb1ebadcdddddc33ebec33de7afb13464849ad49
* :broken_heart: [Build #336287](https://buildkite.com/elastic/kibana-pull-request/builds/336287) failed f9b355ac54ac7b22d74d5ff7d4d6236860e10845
* :broken_heart: [Build #335952](https://buildkite.com/elastic/kibana-pull-request/builds/335952) failed 252bc9b6caf0541c6c6d81368cb1391437f4261a

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/337977","number":337977,"commit":"0d7c1245c38ddd0fd8894121412f4255972c37fa"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/337165","number":337165,"commit":"2f43bf9437336e99f8837aa37a9074ca9e0a534e"},{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/336727","number":336727,"commit":"20d1fe0c6f6f24583a49cc263ced6cdf98b54dc0"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/336574","number":336574,"commit":"fb1ebadcdddddc33ebec33de7afb13464849ad49"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/336287","number":336287,"commit":"f9b355ac54ac7b22d74d5ff7d4d6236860e10845"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/335952","number":335952,"commit":"252bc9b6caf0541c6c6d81368cb1391437f4261a"}],"number":337977}
buildkite-pr-comment-->

denar50 committed:
fix: make Detection Rules visible again

denar50 committed:
WIP

dplumlee committed:
updates v4's to v5's

dplumlee committed:
Merge remote-tracking branch 'upstream/main' into rules-rbac

dplumlee committed:
merge changes

dplumlee committed:
updates feature configs

rylnd committed:
Remove unknown (and probably unneeded) feature field

rylnd committed:
Fix reference to experimental feature

By referencing an unknown flag, we were removing this privilege, which caused errors

rylnd committed:
Update feature authorization snapshots

We're not really sure if this is _correct_ yet, but it's a good
reference for what the system thinks we're trying to do.

elasticmachine commented:
## :broken_heart: Build Failed
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/347282)
* Commit: 355f71e900ee8a316062c600912620558287f21b

### Failed CI Steps


### History
* :broken_heart: [Build #347271](https://buildkite.com/elastic/kibana-pull-request/builds/347271) failed 355f71e900ee8a316062c600912620558287f21b
* :broken_heart: [Build #338341](https://buildkite.com/elastic/kibana-pull-request/builds/338341) failed cfbdc696eec491d60c61a57711313be0393c3859
* :broken_heart: [Build #337977](https://buildkite.com/elastic/kibana-pull-request/builds/337977) failed 0d7c1245c38ddd0fd8894121412f4255972c37fa
* :broken_heart: [Build #337165](https://buildkite.com/elastic/kibana-pull-request/builds/337165) failed 2f43bf9437336e99f8837aa37a9074ca9e0a534e
* :broken_heart: [Build #336727](https://buildkite.com/elastic/kibana-pull-request/builds/336727) failed 20d1fe0c6f6f24583a49cc263ced6cdf98b54dc0
* :broken_heart: [Build #336574](https://buildkite.com/elastic/kibana-pull-request/builds/336574) failed fb1ebadcdddddc33ebec33de7afb13464849ad49

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/347282","number":347282,"commit":"355f71e900ee8a316062c600912620558287f21b"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/347271","number":347271,"commit":"355f71e900ee8a316062c600912620558287f21b"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/338341","number":338341,"commit":"cfbdc696eec491d60c61a57711313be0393c3859"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/337977","number":337977,"commit":"0d7c1245c38ddd0fd8894121412f4255972c37fa"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/337165","number":337165,"commit":"2f43bf9437336e99f8837aa37a9074ca9e0a534e"},{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/336727","number":336727,"commit":"20d1fe0c6f6f24583a49cc263ced6cdf98b54dc0"}],"number":347282}
buildkite-pr-comment-->

dplumlee committed:
Merge branch 'main' into rules-rbac

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/347311)
* Commit: 31bb8e23beff083149f571fd8efeb552c473327f

### Failed CI Steps
* [Linting](https://buildkite.com/elastic/kibana-pull-request/builds/347311#0199aba9-4459-4544-8232-d6f6e43d867f)

### History
* :broken_heart: [Build #347282](https://buildkite.com/elastic/kibana-pull-request/builds/347282) failed 355f71e900ee8a316062c600912620558287f21b
* :broken_heart: [Build #347271](https://buildkite.com/elastic/kibana-pull-request/builds/347271) failed 355f71e900ee8a316062c600912620558287f21b
* :broken_heart: [Build #338341](https://buildkite.com/elastic/kibana-pull-request/builds/338341) failed cfbdc696eec491d60c61a57711313be0393c3859
* :broken_heart: [Build #337977](https://buildkite.com/elastic/kibana-pull-request/builds/337977) failed 0d7c1245c38ddd0fd8894121412f4255972c37fa
* :broken_heart: [Build #337165](https://buildkite.com/elastic/kibana-pull-request/builds/337165) failed 2f43bf9437336e99f8837aa37a9074ca9e0a534e
* :broken_heart: [Build #336727](https://buildkite.com/elastic/kibana-pull-request/builds/336727) failed 20d1fe0c6f6f24583a49cc263ced6cdf98b54dc0

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/347311","number":347311,"commit":"31bb8e23beff083149f571fd8efeb552c473327f"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/347282","number":347282,"commit":"355f71e900ee8a316062c600912620558287f21b"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/347271","number":347271,"commit":"355f71e900ee8a316062c600912620558287f21b"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/338341","number":338341,"commit":"cfbdc696eec491d60c61a57711313be0393c3859"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/337977","number":337977,"commit":"0d7c1245c38ddd0fd8894121412f4255972c37fa"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/337165","number":337165,"commit":"2f43bf9437336e99f8837aa37a9074ca9e0a534e"}],"number":347311}
buildkite-pr-comment-->

dplumlee committed:
fix types

dplumlee committed:
deletes leftover file from earlier merge

elasticmachine commented:
## :broken_heart: Build Failed
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/347336)
* Commit: 5303d4bb51cb3d875ee9d7d92d54bfbaf8ebd93c

### Failed CI Steps
* [Check Types](https://buildkite.com/elastic/kibana-pull-request/builds/347336#0199ac06-1cc1-465e-a8c9-d0e3d976176b)

### History
* :broken_heart: [Build #347328](https://buildkite.com/elastic/kibana-pull-request/builds/347328) failed 7fdd66752bb6ad78dc67da6102f2f32e7d160446
* :broken_heart: [Build #347311](https://buildkite.com/elastic/kibana-pull-request/builds/347311) failed 31bb8e23beff083149f571fd8efeb552c473327f
* :broken_heart: [Build #347282](https://buildkite.com/elastic/kibana-pull-request/builds/347282) failed 355f71e900ee8a316062c600912620558287f21b
* :broken_heart: [Build #347271](https://buildkite.com/elastic/kibana-pull-request/builds/347271) failed 355f71e900ee8a316062c600912620558287f21b
* :broken_heart: [Build #338341](https://buildkite.com/elastic/kibana-pull-request/builds/338341) failed cfbdc696eec491d60c61a57711313be0393c3859

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/347336","number":347336,"commit":"5303d4bb51cb3d875ee9d7d92d54bfbaf8ebd93c"},{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/347328","number":347328,"commit":"7fdd66752bb6ad78dc67da6102f2f32e7d160446"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/347311","number":347311,"commit":"31bb8e23beff083149f571fd8efeb552c473327f"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/347282","number":347282,"commit":"355f71e900ee8a316062c600912620558287f21b"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/347271","number":347271,"commit":"355f71e900ee8a316062c600912620558287f21b"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/338341","number":338341,"commit":"cfbdc696eec491d60c61a57711313be0393c3859"}],"number":347336}
buildkite-pr-comment-->

rylnd committed:
Update lists privileges hook with new privileges hook names

This is something that was missed in the merging of `siemv4` privileges
from upstream.

rylnd committed:
Undo optional field on DataTableModel

This was made optional in the monolithic initial commit on this branch;
it's not clear why. However, this was causing type issues when
attempting to use the field, so I've reverted to the original
definition, and added a TODO comment to investigate why this was
changed in the first place.

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/348524)
* Commit: fcd3f678a36d29c062c67ba9bf149dfb23542015

### Failed CI Steps
* [FTR Configs #127](https://buildkite.com/elastic/kibana-pull-request/builds/348524#0199c1ce-6a46-4059-b79f-0a90a4502ab8)
* [Scout Test Run Builder](https://buildkite.com/elastic/kibana-pull-request/builds/348524#0199c1a9-94ff-4773-8d42-ba27614f39af)
* [Scout Test Run Builder](https://buildkite.com/elastic/kibana-pull-request/builds/348524#0199c1d5-2d67-413a-97ce-43d24bf0d5e1)

### Test Failures
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/348524#0199c1ce-6a46-4059-b79f-0a90a4502ab8) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/348524/jobs/0199c1ce-6a46-4059-b79f-0a90a4502ab8/artifacts/0199c1dc-6a35-489d-bab4-264aac64b751) FTR Configs #<span></span>127 / Endpoint plugin spaces support @<span></span>ess @<span></span>skipInServerless, @<span></span>skipInServerlessMKI Endpoint Artifacts space awareness support for Endpoint Security Trusted Apps List and user does NOT have global artifact management privilege should error if attempting to create a global artifact
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/348524#0199c1a9-94ff-4773-8d42-ba27614f39af) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/348524/jobs/0199c1a9-94ff-4773-8d42-ba27614f39af/artifacts/0199c1d5-19e0-44a9-8ad3-f692023fc2a9) Scout Test Run Builder / serverless-security - EUI testing wrapper: EuiComboBox - with the single selection
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/348524#0199c1d5-2d67-413a-97ce-43d24bf0d5e1) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/348524/jobs/0199c1d5-2d67-413a-97ce-43d24bf0d5e1/artifacts/0199c1df-0fba-4953-bbb3-696d92e366d2) Scout Test Run Builder / serverless-security - EUI testing wrapper: EuiComboBox - with the single selection

### History
* :broken_heart: [Build #347336](https://buildkite.com/elastic/kibana-pull-request/builds/347336) failed 5303d4bb51cb3d875ee9d7d92d54bfbaf8ebd93c
* :broken_heart: [Build #347328](https://buildkite.com/elastic/kibana-pull-request/builds/347328) failed 7fdd66752bb6ad78dc67da6102f2f32e7d160446
* :broken_heart: [Build #347311](https://buildkite.com/elastic/kibana-pull-request/builds/347311) failed 31bb8e23beff083149f571fd8efeb552c473327f
* :broken_heart: [Build #347282](https://buildkite.com/elastic/kibana-pull-request/builds/347282) failed 355f71e900ee8a316062c600912620558287f21b
* :broken_heart: [Build #347271](https://buildkite.com/elastic/kibana-pull-request/builds/347271) failed 355f71e900ee8a316062c600912620558287f21b
* :broken_heart: [Build #338341](https://buildkite.com/elastic/kibana-pull-request/builds/338341) failed cfbdc696eec491d60c61a57711313be0393c3859

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/348524","number":348524,"commit":"fcd3f678a36d29c062c67ba9bf149dfb23542015"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/347336","number":347336,"commit":"5303d4bb51cb3d875ee9d7d92d54bfbaf8ebd93c"},{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/347328","number":347328,"commit":"7fdd66752bb6ad78dc67da6102f2f32e7d160446"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/347311","number":347311,"commit":"31bb8e23beff083149f571fd8efeb552c473327f"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/347282","number":347282,"commit":"355f71e900ee8a316062c600912620558287f21b"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/347271","number":347271,"commit":"355f71e900ee8a316062c600912620558287f21b"}],"number":348524}
buildkite-pr-comment-->

rylnd committed:
Add fleet capabilities to the default client services mock

In this branch we added a new check for fleet capabilities to the global
header component; the fact that this capability was not mocked in many
of our component tests lead to many test failures. For now, I'm setting
this in the default mock and running CI to get a sense for how many jest
tests were affected by this.

There are several cases where we also added these mocks manually in this
PR; hopefully those can be deleted except in cases where we want to use
non-default or specific capability-based behaviors.

denar50 committed:
fix FTR test: add missing permissions to test role

rylnd committed:
Update hook tests for determining missing user privileges

The logic for this hook has been updated to report the missing rules
feature instead of the more general security feature as it did before.
However, I'm not clear whether there are still cases where we want to
report the security feature as missing, so I've added a TODO to follow
up.

rylnd committed:
Merge branch 'main' into rules-rbac

 Conflicts:
	x-pack/solutions/security/plugins/security_solution/public/siem_migrations/links.ts

kibanamachine committed:
[CI] Auto-commit changed files from 'node scripts/eslint_all_files --no-cache --fix'

denar50 committed:
Move product feature detections and external detections to rules
There is one detections still in siem due to CSP not being migrated to rules for the time being.
This commit also removes unused  cloud-defend-* privileges.

kibanamachine committed:
[CI] Auto-commit changed files from 'node scripts/eslint_all_files --no-cache --fix'

denar50 committed:
fix typing error

denar50 committed:
fix test roles
we replaced the hardcoded feature name with SECURITY_FEATURE_ID, however the base object has "siemV3" hardcoded. Since SECURITY_FEATURE_ID is currently "siemV5", some tests using the affected roles break because two versions of the same feature appear at the same time.
Additionally the reference to RULES_FEATURE_ID was removed since the old roles should work without it.

denar50 committed:
fix multiple FTR tests

kibanamachine committed:
[CI] Auto-commit changed files from 'node scripts/eslint_all_files --no-cache --fix'

denar50 committed:
Put back bulkGetUserProfiles inside siemVX detections product

denar50 committed:
Merge branch 'main' into rules-rbac

denar50 committed:
Merge branch 'main' into rules-rbac

denar50 committed:
revert base roles to siemV3
the migration to siemV5 will be handled in a separate follow up PR

denar50 committed:
Merge branch 'main' into rules-rbac

denar50 committed:
fix broken FTR

denar50 committed:
fix broken FTR

denar50 committed:
fix broken cypress test due to a change in the ordering of siem sub features

denar50 committed:
fix broken rules management cypress test
it was relying on being able to navigate to the add rules page even when the user did not have permissions to do so. Before this used to work because the page was added when the user had access to either detections or external_detections, but the RBAC pr changed this to explicitly require rules write access.

denar50 committed:
fix AI4SOC cypress test

denar50 committed:
revert quotation change in roles file

denar50 committed:
Merge branch 'main' into rules-rbac

denar50 committed:
Merge branch 'main' into rules-rbac

denar50 committed:
fix ftr tests after trusted devices was enabled

denar50 committed:
fix jest tests

kibanamachine committed:
[CI] Auto-commit changed files from 'node scripts/eslint_all_files --no-cache --fix'

elasticmachine commented:
## :broken_heart: Build Failed
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/352233)
* Commit: 7ebca83a15df37214d123527d9934c22696b5c76

### Failed CI Steps
* [Linting](https://buildkite.com/elastic/kibana-pull-request/builds/352233#0199f292-edc1-4a85-a44e-30be1d487004)

### History
* :broken_heart: [Build #352089](https://buildkite.com/elastic/kibana-pull-request/builds/352089) failed 74674ecf6410ddf5ec675459215467965e997dd1
* :broken_heart: [Build #351767](https://buildkite.com/elastic/kibana-pull-request/builds/351767) failed 1d374bda1eb4e8884dae974fc1c2a4fcc22d76da
* :broken_heart: [Build #351237](https://buildkite.com/elastic/kibana-pull-request/builds/351237) failed 73b5471517ef1e5868c2ee925fcbbc65be7ee5c0
* :broken_heart: [Build #350771](https://buildkite.com/elastic/kibana-pull-request/builds/350771) failed 021e3949857945c0638b036a2694c2e06ba6aec1
* :broken_heart: [Build #350462](https://buildkite.com/elastic/kibana-pull-request/builds/350462) failed e3bdd10cad62ea21dfbf1a5f64a79d562a4d5265

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/352233","number":352233,"commit":"7ebca83a15df37214d123527d9934c22696b5c76"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/352089","number":352089,"commit":"74674ecf6410ddf5ec675459215467965e997dd1"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/351767","number":351767,"commit":"1d374bda1eb4e8884dae974fc1c2a4fcc22d76da"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/351237","number":351237,"commit":"73b5471517ef1e5868c2ee925fcbbc65be7ee5c0"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/350771","number":350771,"commit":"021e3949857945c0638b036a2694c2e06ba6aec1"}],"number":352233}
buildkite-pr-comment-->

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/352278)
* Commit: 46fa01b551839554e21ad728c800b1157a63895a
* [Storybooks Preview](https://ci-artifacts.kibana.dev/storybooks/pr-232113)

### Failed CI Steps
* [Jest Tests #12](https://buildkite.com/elastic/kibana-pull-request/builds/352278#0199f33b-bf89-46bb-9c47-04dbfbe6d6c7)
* [FTR Configs #28](https://buildkite.com/elastic/kibana-pull-request/builds/352278#0199f33b-bfa4-4230-a839-c48733ef7db8)
* [FTR Configs #28](https://buildkite.com/elastic/kibana-pull-request/builds/352278#0199f365-bda6-4a7a-8c91-13de304bb0dd)
* [FTR Configs #116](https://buildkite.com/elastic/kibana-pull-request/builds/352278#0199f33b-bfea-42ce-bc56-d256888101ee)

### Test Failures
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/352278#0199f33b-bfea-42ce-bc56-d256888101ee) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/352278/jobs/0199f33b-bfea-42ce-bc56-d256888101ee/artifacts/0199f366-f991-4b43-b093-167b20ef2bd4) FTR Configs #<span></span>116 / Alerts APIs - Trial License/Complete Tier Alert assignments API @<span></span>serverless Alert User Assignment - Serverless authorization / RBAC should allow `ROLES.t1_analyst` to assign alerts
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/352278#0199f33b-bfa4-4230-a839-c48733ef7db8) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/352278/jobs/0199f33b-bfa4-4230-a839-c48733ef7db8/artifacts/0199f365-6e49-4eea-b3ce-8acf8e5650e1) FTR Configs #<span></span>28 / security (trial license) Privileges GET /api/security/privileges should return a privilege map with all known privileges, without actions
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/352278#0199f365-bda6-4a7a-8c91-13de304bb0dd) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/352278/jobs/0199f365-bda6-4a7a-8c91-13de304bb0dd/artifacts/0199f36b-88df-4222-aecc-a8df98a7461d) FTR Configs #<span></span>28 / security (trial license) Privileges GET /api/security/privileges should return a privilege map with all known privileges, without actions
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/352278#0199f33b-bf89-46bb-9c47-04dbfbe6d6c7) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/352278/jobs/0199f33b-bf89-46bb-9c47-04dbfbe6d6c7/artifacts/0199f357-115a-4e09-ac5c-6a0eab2b9fc7) Jest Tests #<span></span>12 / useAlertExceptionActions should disable adding endpoint exceptions when alert is not an endpoint alert
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/352278#0199f33b-bf89-46bb-9c47-04dbfbe6d6c7) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/352278/jobs/0199f33b-bf89-46bb-9c47-04dbfbe6d6c7/artifacts/0199f357-115b-4372-a5f2-af2d4264abd4) Jest Tests #<span></span>12 / useAlertExceptionActions should disable adding endpoint exceptions when user has no endpoint exceptions ALL privilege
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/352278#0199f33b-bf89-46bb-9c47-04dbfbe6d6c7) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/352278/jobs/0199f33b-bf89-46bb-9c47-04dbfbe6d6c7/artifacts/0199f357-115a-4797-b403-2df1c56c46c6) Jest Tests #<span></span>12 / useAlertExceptionActions should disable adding rule exceptions when user has no index write privilege
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/352278#0199f33b-bf89-46bb-9c47-04dbfbe6d6c7) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/352278/jobs/0199f33b-bf89-46bb-9c47-04dbfbe6d6c7/artifacts/0199f357-115c-453c-adc7-f9c903153d38) Jest Tests #<span></span>12 / useAlertExceptionActions should disable adding rule exceptions when user has no security:ALL privilege
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/352278#0199f33b-bf89-46bb-9c47-04dbfbe6d6c7) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/352278/jobs/0199f33b-bf89-46bb-9c47-04dbfbe6d6c7/artifacts/0199f357-115a-4b47-b7d2-5f4576a4f41e) Jest Tests #<span></span>12 / useAlertExceptionActions should return both add rule exception and add endpoint exception menu items with all privileges

### History
* :broken_heart: [Build #352233](https://buildkite.com/elastic/kibana-pull-request/builds/352233) failed 7ebca83a15df37214d123527d9934c22696b5c76
* :broken_heart: [Build #352089](https://buildkite.com/elastic/kibana-pull-request/builds/352089) failed 74674ecf6410ddf5ec675459215467965e997dd1
* :broken_heart: [Build #351767](https://buildkite.com/elastic/kibana-pull-request/builds/351767) failed 1d374bda1eb4e8884dae974fc1c2a4fcc22d76da
* :broken_heart: [Build #351237](https://buildkite.com/elastic/kibana-pull-request/builds/351237) failed 73b5471517ef1e5868c2ee925fcbbc65be7ee5c0
* :broken_heart: [Build #350771](https://buildkite.com/elastic/kibana-pull-request/builds/350771) failed 021e3949857945c0638b036a2694c2e06ba6aec1

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/352278","number":352278,"commit":"46fa01b551839554e21ad728c800b1157a63895a"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/352233","number":352233,"commit":"7ebca83a15df37214d123527d9934c22696b5c76"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/352089","number":352089,"commit":"74674ecf6410ddf5ec675459215467965e997dd1"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/351767","number":351767,"commit":"1d374bda1eb4e8884dae974fc1c2a4fcc22d76da"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/351237","number":351237,"commit":"73b5471517ef1e5868c2ee925fcbbc65be7ee5c0"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/350771","number":350771,"commit":"021e3949857945c0638b036a2694c2e06ba6aec1"}],"number":352278}
buildkite-pr-comment-->

denar50 committed:
fix FTR test
a t1_analyst with read only access should be able to assign alerts

denar50 committed:
fix FTR test

denar50 committed:
fix unit test

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/352310)
* Commit: e05f62b199e3e522562b8c914dcc4b97fbb45872

### Failed CI Steps
* [Linting](https://buildkite.com/elastic/kibana-pull-request/builds/352310#0199f397-dee1-4f98-bb19-85fa527bf4e9)

### History
* :broken_heart: [Build #352278](https://buildkite.com/elastic/kibana-pull-request/builds/352278) failed 46fa01b551839554e21ad728c800b1157a63895a
* :broken_heart: [Build #352233](https://buildkite.com/elastic/kibana-pull-request/builds/352233) failed 7ebca83a15df37214d123527d9934c22696b5c76
* :broken_heart: [Build #352089](https://buildkite.com/elastic/kibana-pull-request/builds/352089) failed 74674ecf6410ddf5ec675459215467965e997dd1
* :broken_heart: [Build #351767](https://buildkite.com/elastic/kibana-pull-request/builds/351767) failed 1d374bda1eb4e8884dae974fc1c2a4fcc22d76da
* :broken_heart: [Build #351237](https://buildkite.com/elastic/kibana-pull-request/builds/351237) failed 73b5471517ef1e5868c2ee925fcbbc65be7ee5c0

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/352310","number":352310,"commit":"e05f62b199e3e522562b8c914dcc4b97fbb45872"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/352278","number":352278,"commit":"46fa01b551839554e21ad728c800b1157a63895a"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/352233","number":352233,"commit":"7ebca83a15df37214d123527d9934c22696b5c76"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/352089","number":352089,"commit":"74674ecf6410ddf5ec675459215467965e997dd1"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/351767","number":351767,"commit":"1d374bda1eb4e8884dae974fc1c2a4fcc22d76da"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/351237","number":351237,"commit":"73b5471517ef1e5868c2ee925fcbbc65be7ee5c0"}],"number":352310}
buildkite-pr-comment-->

denar50 committed:
fix linting issues

rylnd commented:
Closing in favor of https://github.com/elastic/kibana/pull/239634.
